### PR TITLE
Grab Bag

### PR DESCRIPTION
**Pull request by systmkor**
Before I get too far away from the original code. I figured I would make a pull request.
# Main Changes
- Broke up code into submodules accordingly
- Created a requirements.txt file
- Added basic socks5 support, it only tries connect to default tor address (localhost, 9050)
- Moved CLI argument/option handling from hand made to docopt compatible
- Using python logging module facility instead of just print statements for verbose mode, errors, etc.
# Much To Be Done

I have some ideas of what to add. This isn't everything but some of the main tasks to be done.
- I personally would like to clean up and clarify more of the code
- Make socks/proxy options more configurable
- Write unit tests
- Database Support (SQLlite3, PostgreSQL, some NoSQL database idk specifically)

### CODE REVIEW

**Comment by corelanc0d3r** [comment]
thanks for the major refactor.  I ran a couple of tests and found a couple of issues:
1. when no config file is specified (-c), the app throws an error (because it is still trying to read the config file)
2. I don't see any output unless I add the -v parameter.  It would be nice to at least see basic output (and verbose output when using -v)
3. We'll need to update the installation instructions, (include installation of docopts and PySocks) 
4. The email report doesn't show Host/Port/IP info.  (all fields are set to "None")

**Comment by systmkor** [comment]
**tl;dr** I can fix all of the issues except for 2. 

I would prefer that I fix the issues you noted (i.e. except 2); then we discuss or you choose which logging statements should be: regular, verbose, warning, error, or debug. Let me know if this is a bad plan or a better way. :smile:

**Comment by corelanc0d3r** [comment]
not a bad idea to implement different verbose logging levels, but I would prefer to have basic output by default  (and, if possible, not using the python logging module as it doesn't improve readability imho).  Anyone running the tool with default settings should at least see basic results  (and not wonder if they forgot something or of the tool is working or not)

**Comment by systmkor** [comment]
Okay, that's easy to change with the python logging module. Also you can change the formatting of python logging module so it's easier to read for default output and for debugging output have a different formatting style.
